### DISCUSSION TITLE
Can't open excel or powerpoint files in edit mode

### DISCUSSION THREAD

**Discussion by sf360bnedeljkovic** [petrsvihlik/WopiHost#169] [Q&A] [answered]
Hi,

I found your project and I am trying to adapt it to my usecase (we have a .net 4.8 web app), so I had to downgrade and rework code to make it work with .net framework...

I managed to get DOCX files to properly open and save in edit mode, however that's not the case for excel or powerpoint...  With excel I get **"We're sorry, we couldn't open your workbook. It's possibly corrupt or using a file format that's not supported."**

I have compared your host and min - I can see that Files/Id is sending correct JSON response to OOS (at least I don't see anything different), then it triggers Fies/ProcessLock couple of time it stops (and it returns OK), and then it stops... 

ULS logs proved less than useless (I have enabled verbose logging), but I can't pinpoint the mistake I have....

Could you please help me in how I can debug the error or how to get something out of OOS server?

**Comment by sf360bnedeljkovic** [comment]
Update: Just to eliminate any issues coming from framework, I moved my changes to your source and still can't open pptx/xlsx in edit mode. 

Our files in storage don't have extension in its name (I set this myself in CheckFileInfo), could this be the culprit?

Update2: It is definitely file extension... When I appended .xlsx into blob name editor was able to open it. Is there a way to do this without file extension? We have hundreds of thousands of blob files in our storage without extension (we keep track of extensions via different means)?

**Reply by petrsvihlik** [reply]
You're supposed to return an implementation of the `IWopiFile` in your implementation of the `IWopiStorageProvider.GetWopiFile`. How you implement the `IWopiFile.Name` property is up to you. I suppose you have your extensions stored in the blob somewhere as metadata, so append it to the filename and return it in the `Name` property.

I didn't test WOPI with extensionless files, but I can imagine WOPI clients such as OWA or OOS being sensitive to extensions.

**Reply by sf360bnedeljkovic** [reply, marked as answer]
Hi Petr,

I managed to find the root cause of the issue -  **CheckFileInfo.BaseFileName** property is required to have extension (and I was only setting our blob name here).

For DOCX endpoint, this is not important, but for PPTX/XLSX it is... In all fairness, [MS does say that extension is required here](https://learn.microsoft.com/en-us/microsoft-365/cloud-storage-partner-program/rest/files/checkfileinfo/checkfileinfo-response#basefilename), however they do not make it easy to identify the problem.
